### DISCUSSION TITLE
AntMedia Web Player and Data Channel

### DISCUSSION THREAD

**Discussion by MIDIculous** [ant-media/Ant-Media-Server#6637] [General]
Does the AntMedia [Web-Player](https://github.com/ant-media/Web-Player) support the WebRTC Data Channel? Use-Case is a 1:1 student and teacher and the teacher wants to send the student custom images, or text, etc....

How would this be achieved?

**Comment by lastpeony** [comment]
Hello,
Yes it does support WebRTC Data Channel.

You can pass webRTCDataListener function to web players constructor to listen incoming data:

```
function webRtcDataReceiver(data){
console.log("webrtc data received: "+data)
}

var player = new WebPlayer({
  streamId: "myStreamId",
  webRTCDataListener: webRtcDataReceiver,
  httpBaseURL: "http://example.antmedia.io:5080/WebRTCAppEE/", 
}, document.getElementById("video_container"), null);
```
Alternatively you can use `addWebRTCDataListener `method to add listener.
Example:
`player.addWebRTCDataListener(webRtcDataReceiver);
`

To send a data channel message use `sendWebRTCData `method.

Example:
```
var data ={ someKey:"someValue"}
player.sendWebRTCData(data)
```
Behind the scenes this will call javascript sdks `webRTCAdaptor.sendData` function.

**Comment by MIDIculous** [comment]
@lastpeony

Thanks for this! Appreciate it!
